client/packages.go

```diff
@@ -40,6 +40,8 @@ type Snap struct {
 	Type          string    `json:"type"`
 	Version       string    `json:"version"`
 	Revision      Revision  `json:"revision"`
+	Confinement   string    `json:"confinement"`
```

Comment:
Yes, I think the `devmode` as reported from `snap find` should be the `Confinement` attribute, whereas the `devmode` as reported from `snap list` should be from `SnapStateFlags` (or `SnapSetup`; still looking in to which of those is relevant and/or easier to get at).

This branch just exposes `Confinement` in both places, which is probably not what the user wants to know for the `snap list` case.